### PR TITLE
[46067] Fix English language displayed multiple times in available languages

### PR DESCRIPTION
**Pull request by cbliard**
https://community.openproject.org/wp/46067

The language name displayed in user settings comes from translated files, using the `:general_lang_name` key. If no translations exist for some languages, like Belarusian and Basque languages, it falls back to displaying the English version: "English".

This PR aims at providing translations for all languages automatically.

### CODE REVIEW

File: spec/helpers/application_helper_spec.rb [hunk 1]
```diff
@@ -175,4 +172,48 @@
...
+        The locales #{impostor_locales.to_sentence} display themselves as "English"!
+
+        Probably because new languages were added, and the translation for their language is not
+        available, so it fallbacks to the English translation.
+
+        To fix it, generate translation files from CLDR by running
+
+            script/i18n/generate_languages_translations
+
+        And commit the yml files added in "config/locales/generated/fr.yml".
```
Comment (dombesz) [inline spec/helpers/application_helper_spec.rb]: I assume you wanted to write `"config/locales/generated/*.yml"` ?

File: spec/helpers/application_helper_spec.rb [hunk 2]
```diff
@@ -175,4 +172,48 @@
...
+        all_lang_options_for_select(false)
+          .filter { |lang, _locale| duplicate_langs.include?(lang) }
+          .sort
+
+      expect(duplicate_options.count).to eq(0), <<~ERR
+        Some identical language names are used for different locales!
+
+          duplicates: #{duplicate_options}
+
+        To fix it, inspect translation files located in "config/locales/generated/fr.yml".
```
Comment (dombesz) [inline spec/helpers/application_helper_spec.rb]: Same, `*.yml`?

**Review by dombesz — APPROVED** [review]
LGTM! Just two typos spotted. Feel free to merge once those questions are cleared.
Thanks @cbliard!

File: spec/helpers/application_helper_spec.rb (hunk 1 above)
Comment (cbliard) [reply spec/helpers/application_helper_spec.rb]: yes! Thanks for spotting

File: spec/helpers/application_helper_spec.rb (hunk 2 above)
Comment (cbliard) [reply spec/helpers/application_helper_spec.rb]: same

**Pull request merged**